virtual_ecosystem/models/litter/env_factors.py

```diff
@@ -5,6 +5,88 @@
 
 import numpy as np
 from numpy.typing import NDArray
+from xarray import DataArray
+
+from virtual_ecosystem.core.core_components import LayerStructure
+from virtual_ecosystem.models.litter.constants import LitterConsts
+
+
+def calculate_environmental_factors(
+    air_temperatures: DataArray,
+    soil_temperatures: DataArray,
+    water_potentials: DataArray,
+    layer_structure: LayerStructure,
+    constants: LitterConsts,
+):
+    """Calculate the impact of the environment has on litter decay across litter layers.
+
+    For the above ground layer the impact of temperature is calculated, and for the
+    below ground layer the effect of temperature and soil water potential are both
+    calculated.
+
+    The relevant above ground temperature is the surface temperature, which can be
+    easily extracted from the temperature data. It's more complex for the below ground
+    temperature and the water potential as the relevant values are averages across the
+    microbially active depth. These are calculated by averaging across the soil layers
+    with each layer weighted by the proportion of the total microbially active depth it
+    represents.
+
+    If a shallow microbially active depth is used then below ground litter decomposition
+    will be exposed to a high degree of environmental variability. This is
+    representative of the real world, but needs to be kept in mind when comparing to
+    other models.
+
+    Args:
+        air_temperatures: Air temperatures, for all above ground layers [C]
+        soil_temperatures: Soil temperatures, for all soil layers [C]
+        water_potentials: Water potentials, for all soil layers [kPa]
+        layer_structure: The LayerStructure instance for the simulation.
+        constants: Set of constants for the litter model
+
+    Returns:
+        A dictionary containing three environmental factors, one for the effect of
+        temperature on above ground litter decay, one for the effect of temperature on
+        below ground litter decay, and one for the effect of soil water potential on
+        below ground litter decay.
+    """
+
+    temperatures = {
+        "surface": air_temperatures[layer_structure.index_surface_scalar].to_numpy(),
```

Comment:
Yep this one is for the above ground litter, in this case I just use the surface temperature directly and don't interpolate (because that seems tricky without tracking litter depth etc)